server.js

```diff
@@ -1,36 +1,118 @@
+import booksData from './data/books.json'
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/books"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// Defines the port the app will run on
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(express.json())
 
-// Start defining your routes here
+// middleware that checks if database is connected before going to the endpoints
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+// seed the database with data of books
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({})
+
+    booksData.forEach(book => {
+      new Book(book).save()
+    })
+  }
+
+  seedDatabase()
+}
+
+// main endpoint
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// list of endpoints
+app.get('/endpoints', (req, res) => {
+  res.send(listEndpoints(app))
+})
+
+// endpoint to get all the books
+app.get('/books', async (req, res) => {
+  const books = await Book.find()
+  res.json(books)
+})
+
+// endpoint to get a specific book based on id
+app.get('/books/:id', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const bookById = await Book.findById(id)
+
+    if (!bookById) {
+      res.status(404).json({
+        response: 'No book found with that id',
+        success: false
+      })
+    } else {
+      res.status(200).json({
+        response: bookById,
+        success: true
+      })
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Id is invalid'})
+  }
+})
+
+// endpoint to get a specific book based on title
+app.get('/books/title/:title', async (req, res) => {
+  const { title } = req.params
+
+  try {
+    const bookTitle = await Book.find(title)
+  
```

Comment:
Whenever I enter a title in postman, it throws the error `'Title is invalid'`. Try and use an object for the search, like so: 
`const bookTitle = await Book.find({ title })`
🙂 Does that work?